hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/WrappedWritableStates.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.spi.state.WrappedWritableKVState;
+import com.hedera.node.app.spi.state.WrappedWritableSingletonState;
+import com.hedera.node.app.spi.state.WritableKVState;
+import com.hedera.node.app.spi.state.WritableQueueState;
+import com.hedera.node.app.spi.state.WritableSingletonState;
+import com.hedera.node.app.spi.state.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A wrapper around a {@link WritableStates} that provides a wrapped instances of {@link WritableKVState} and
+ * {@link WritableSingletonState} of a given {@link WritableStates} delegate.
+ */
+public class WrappedWritableStates implements WritableStates {
+
+    private final WritableStates delegate;
+    private final Map<String, WrappedWritableKVState<?, ?>> writableKVStateMap = new HashMap<>();
+    private final Map<String, WrappedWritableSingletonState<?>> writableSingletonStateMap = new HashMap<>();
```

Comment:
We will need to add WrappedWritableQueueState into here.

Reply:
Created https://github.com/hashgraph/hedera-services/issues/6762 to track this.